core-bundle/src/Image/Studio/ImageResult.php

```diff
@@ -142,6 +145,38 @@ public function getFilePath($absolute = false): string
         return $absolute ? $path : Path::makeRelative($path, $this->projectDir);
     }
 
+    /**
+     * Synchronously process images if they are deferred.
+     *
+     * This will make sure that the target files physically exist instead of
+     * being generated by the Contao\CoreBundle\Controller\ImagesController
+     * on first access.
+     */
+    public function createIfDeferred(): void
+    {
+        $resizer = $this->resizer();
+
+        if (!$resizer instanceof DeferredResizerInterface) {
+            throw new \RuntimeException('The "contao.image.resizer" service does not support deferred resizing.');
+        }
```

Comment:
I think it should only throw if `$deferredImages` is not empty. If all images are already resized throwing an exception here doesn’t have any benefit, right?

Reply:
My reasoning was: now you'll get noticed about a wrong usage even if the files are already processed. Not throwing would kind of hide this in some cases and you would probably only stumble across the error in production. :thinking:  

But I'm not 100% sure how a setup would look if you both have 'deferred' images + a resizer that does not support them.

Reply:
Having deferred images but not a deferred resizer would be a very special edge case. But if I want to disable deferred images in my setup completley (wich is a valid use case IMO) I still want `->createIfDeferred()` to work and not throw exceptions.

Reply:
That's a valid point. I adjusted the behavior in b73ba710ac2965c1a0137b765101e7c9c797ad5d so that an exception will only get thrown if there *are* deferred images present.